Windows (MSVC) build: in_addr_t undeclared in udp_transport.cpp

## Description

Building opensomeip with MSVC (Visual Studio 17 2022) on Windows fails because `in_addr_t` is not defined. This is a POSIX type (`<netinet/in.h>`) that does not exist on Windows/Winsock2.

Previous portability fixes (PR #70) resolved most issues, but `in_addr_t` was missed in the `is_multicast_address()` function.

## Error

```
udp_transport.cpp(460,5):  error C2065: 'in_addr_t': undeclared identifier
udp_transport.cpp(460,15): error C2146: syntax error: missing ';' before identifier 'addr'
udp_transport.cpp(460,15): error C2065: 'addr': undeclared identifier
udp_transport.cpp(461,9):  error C2065: 'addr': undeclared identifier
udp_transport.cpp(466,32): error C2065: 'addr': undeclared identifier
```

CI run: https://github.com/vtz/opensomeip-python/actions/runs/23021889413

## Root cause

In `src/transport/udp_transport.cpp`, line 460:

```cpp
bool UdpTransport::is_multicast_address(const std::string& address) const {
    in_addr_t addr = inet_addr(address.c_str());  // <-- in_addr_t does not exist on Windows
    if (addr == INADDR_NONE) {
        return false;
    }
    uint32_t host_addr = ntohl(addr);
    return (host_addr >= 0xE0000000) && (host_addr <= 0xEFFFFFFF);
}
```

`in_addr_t` is defined as `uint32_t` in POSIX `<netinet/in.h>`, but Windows/Winsock2 does not provide this typedef.

## Suggested fix

Add a typedef to `include/platform/win32/net_impl.h`:

```cpp
using in_addr_t = u_long;  // inet_addr() returns u_long on Windows
```

Or replace the usage directly:

```cpp
unsigned long addr = inet_addr(address.c_str());
```

Both `inet_addr()` return type and `INADDR_NONE` are already compatible with `unsigned long` on Windows.

## Environment

- **Compiler**: MSVC 17.14 (Visual Studio 17 2022)
- **OS**: Windows Server 2022 (GitHub Actions `windows-latest`)
- **Python**: 3.10 - 3.13

--- a/include/platform/win32/net_impl.h
+++ b/include/platform/win32/net_impl.h
@@ -13,6 +13,7 @@
 #pragma comment(lib, "ws2_32.lib")
 
 using ssize_t = SSIZE_T;
+using in_addr_t = u_long;
 
 #define someip_close_socket(fd) closesocket(fd)
 #define someip_shutdown_socket(fd) shutdown(fd, SD_BOTH)

